### PR TITLE
feat: single file diff

### PR DESCRIPTION
**Pull request by xyc**
### What does this PR do?
Single file diff using cache service lib

### What issues does this PR fix or reference?
@W-9166550@

### CODE REVIEW

File: packages/salesforcedx-vscode-core/package.json
Comment (jag-sfdc) [inline packages/salesforcedx-vscode-core/package.json:190]: Why are we making this change? Just a cleanup or is there any benefit to this?

File: packages/salesforcedx-vscode-core/src/commands/forceSourceDiff.ts
Comment (jag-sfdc) [inline packages/salesforcedx-vscode-core/src/commands/forceSourceDiff.ts:31]: I refactored this to conflictDetectionService to keep things cleaner here. Let me know if that sounds good once we merge ours.

File: packages/salesforcedx-vscode-core/package.json
Comment (xyc) [reply packages/salesforcedx-vscode-core/package.json:190]: MetadataCacheService allows us to diff more file types, example: `js-meta.xml`

File: packages/salesforcedx-vscode-core/src/commands/forceSourceDiff.ts
Comment (xyc) [reply packages/salesforcedx-vscode-core/src/commands/forceSourceDiff.ts:31]: yeah, that looks good

**Review by jag-sfdc — APPROVED** [review]

File: packages/salesforcedx-vscode-core/src/messages/i18n.ja.ts
Comment (jag-sfdc) [inline packages/salesforcedx-vscode-core/src/messages/i18n.ja.ts:499]: Can you include Sonal and get her approval for this. Rest looks good.

**Comment by AnanyaJha** [comment]
If a user doesn't have a file in their org, we're currently displaying two error notifications: 
Screen Shot 2021-05-17 at 5 01 10 PM

Do we need to display two? Is there a way we could consolidate the info to one pop up for the user?

File: packages/salesforcedx-vscode-core/package.json
Comment (AnanyaJha) [reply packages/salesforcedx-vscode-core/package.json:190]: This is going to cause the `Diff` command to be exposed for across most contexts ie. a user's `.forceignore`, `.gitignore`, the `manifest` folder etc. Is there anyway to keep this more selective, ie. to just files/directories in the project's source folders?

**Comment by xyc** [comment]
@AnanyaJha this is the current pattern of error messages. The text notification_unsuccessful_execution_text doesn't allow us to customize with additional messaging.

File: packages/salesforcedx-vscode-core/src/messages/i18n.ts
Comment (xyc) [inline packages/salesforcedx-vscode-core/src/messages/i18n.ts:565]: @sbudhirajadoc Could I have a review on the text? This is shown when the remote file diffed is not found.

**Comment by AnanyaJha** [comment]
> @AnanyaJha this is the current pattern of error messages. The text notification_unsuccessful_execution_text doesn't allow us to customize with additional messaging.

Ah I see, in that case maybe we should just log the message in the output window so we follow the pattern we're using for other commands? Included an example of an error while deploying 

Screen Shot 2021-05-17 at 5 25 04 PM

File: packages/salesforcedx-vscode-core/package.json
Comment (xyc) [reply packages/salesforcedx-vscode-core/package.json:190]: not really, the source folders can be arbitrary. Also we are keeping this consistent with sdr commands. (Deploy/Retreive commands are also available for the aforementioned files)

**Comment by xyc** [comment]
I'd argue that is inferior UX since it shows the error far away & it takes much more effort for users to locate the root cause. Consolidate messages is not a bad idea, but that would take a much larger refactoring.

File: packages/salesforcedx-vscode-core/src/messages/i18n.ja.ts
Comment (sbudhirajadoc) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ja.ts:499]: looks good to me.

File: packages/salesforcedx-vscode-core/src/messages/i18n.ts
Comment (sbudhirajadoc) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ts:565]: change to 'Selected file or directory was not found in the org'

File: packages/salesforcedx-vscode-core/src/messages/i18n.ja.ts
Comment (sbudhirajadoc) [reply packages/salesforcedx-vscode-core/src/messages/i18n.ja.ts:499]: change to 'Selected file or directory was not found in the org'

**Review by sbudhirajadoc — COMMENTED** [review]
minor changes in text.

**Review by rcoringrato-sfdc — APPROVED** [review]

**Pull request merged**
